### PR TITLE
Adding App Insights resource creation to infrastructure scripts

### PR DESCRIPTION
**Pull request by NarmathaBala**
This PR is to add Application Insights resource creation to IAC into RetrospectiveExtension.Backend folder. 

Note: 
1. This is an initial PR to get the resource creation into scripts. We need to understand how the continuous deployment is going to be structured and evaluate the location of this script.
2. The instrumentation key is collected into a variable but not written anywhere like into console logs

### CODE REVIEW

File: RetrospectiveExtension.Backend/deploy/env_setup.sh [hunk 1]
```diff
@@ -86,20 +101,26 @@
         --name "app-${resource_name_suffix}" \
         --settings Azure__SignalR__ConnectionString=$signalr_connection_string "@dev_certs.json" "@allowed_origins.json"
 
+    # Create Application Insights AppSettings
```
Comment (mhassaninmsft) [inline RetrospectiveExtension.Backend/deploy/env_setup.sh]: I think setting the AI key should be done with the rest of the settings at line 102

File: RetrospectiveExtension.Backend/deploy/env_setup.sh [hunk 2]
```diff
@@ -86,20 +101,26 @@
...
+    az webapp config appsettings set \
+        --resource-group "$resource_group" \
+        --name "ai-${resource_name_suffix}" \
+        --settings Azure__AppInsights__ConnectionString=$ai_instrumentation_key "@dev_certs.json" "@allowed_origins.json"
```
Comment (mhassaninmsft) [inline RetrospectiveExtension.Backend/deploy/env_setup.sh]: The name of the application insights settings should be `ApplicationInsights__InstrumentationKey` since it should match what the current backend code expects

File: RetrospectiveExtension.Frontend/config/environment.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 const environment = {
-  CollaborationStateServiceUrl: '<collaboration state service url>',
+  CollaborationStateServiceUrl: 'https://app-nbrtx.azurewebsites.net/',
```
Comment (mhassaninmsft) [inline RetrospectiveExtension.Frontend/config/environment.tsx]: my current PR will add `enviornment.tsx` to the `.gitignore` list and create a templat file. This waywe do not accidentally upload secrets or other config values

File: RetrospectiveExtension.Frontend/package.json [hunk 4]
```diff
@@ -4,13 +4,13 @@
   "description": "Azure DevOps Retrospective Extension",
   "main": "index.tsx",
   "scripts": {
-    "build:d": "rm -rf ./dist && webpack --mode=development",
-    "build:p": "rm -rf ./dist && webpack --mode=production",
```
Comment (mhassaninmsft) [inline RetrospectiveExtension.Frontend/package.json]: why did you remove build:d and build:p and opted for a single build

File: RetrospectiveExtension.Frontend/vss-extension-dev.json [hunk 5]
```diff
@@ -1,11 +1,12 @@
 {
   "manifestVersion": 1,
   "id": "retrospective-vsts-extension-dev",
-  "publisher": "<publisher id>",
-  "version": "0.0.1",
+  "publisher": "nabala",
```
Comment (mhassaninmsft) [inline RetrospectiveExtension.Frontend/vss-extension-dev.json]: same those will be .gitignored in my upcoming PR

File: RetrospectiveExtension.Frontend/webpack.config.js [hunk 6]
```diff
@@ -8,11 +8,18 @@ const BUILD_DIR = path.resolve(__dirname, 'dist');
 const APP_DIR = path.resolve(__dirname, '');
 
 module.exports = {
-  devtool: 'source-map',
```
Comment (mhassaninmsft) [inline RetrospectiveExtension.Frontend/webpack.config.js]: My guess is that if we pack the extension in production mode, the devServer will not be picked up. Can you verify this. Also this change can change the behavior of the production extension. @polatengin  Do you have any experience with this.
I am lenient towards the safer route and leaving this file unchanged until we verify that it is safe

**Review by mhassaninmsft — CHANGES_REQUESTED** [review]
looks good but there are a couple of issues we need to close I think before merging

**Comment by phongcao** [comment]
@NarmathaBala Please update your branch with `dev`.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -183,9 +183,11 @@ secrets. Remember to increment the name index to add additional secrets.
 1. Once the script completes, it will output the url of the backend service. You can navigate to the [Azure Portal](https://portal.azure.com)
 and validate that the `rg-<RESOURCE_NAME_SUFFIX>` resource group exists and
 contains the App Service, App Service Plan and SignalR resources.
-1. You will need to update the `RetrospectiveExtension.FrontEnd/config/environment.tsx`
-CollaborationStateServiceUrl value to the App Service URL -
-`https://<RESOURCE_NAME_SUFFIX>.azurewebsites.net` and redeploy the extension.
+1. Update the `RetrospectiveExtension.FrontEnd/config/environment.tsx` to reflect changes to:
+   - `CollaborationStateServiceUrl` value to the App Service URL -
+`https://<RESOURCE_NAME_SUFFIX>.azurewebsites.net`.
```
Comment (mhassaninmsft) [inline CONTRIBUTING.md:188]: it is `<app-{RESOURCE_NAME_SUFFIX}>`

File: RetrospectiveExtension.Backend/deploy/env_setup.sh [hunk 8]
```diff
@@ -62,7 +62,22 @@
     az appservice plan create \
         --name "plan-${resource_name_suffix}" \
         --resource-group "$resource_group"
-    
+
+    # https://docs.microsoft.com/en-us/cli/azure/monitor/app-insights/component?view=azure-cli-latest#az-monitor-app-insights-component-create
+    # Create Application Insights resource
+    echo "### Creating Application Insights resource"
```
Comment (mhassaninmsft) [inline RetrospectiveExtension.Backend/deploy/env_setup.sh:68]: Add Trailing #### so that it looks consistent with the rest of the echo statements in the script  for example
`echo "#### Publishing backend project ####"`

**Review by mhassaninmsft — APPROVED** [review]
a couple of minor cosmetic changes, but otherwise looks solid
